File: backend/app/schemas/prediction.py
from pydantic import BaseModel, ConfigDict, Field


class PredictionRequest(BaseModel):
    symptoms: list[str] = Field(min_length=1, max_length=25)


class DiseasePrediction(BaseModel):
    disease: str
    confidence: float
    probability: float
    similarity: float
    final_score: float
    matched_symptoms: list[str]
    missing_symptoms: list[str]


class PredictionResponse(BaseModel):
    predictions: list[DiseasePrediction]

    model_config = ConfigDict(from_attributes=True)

File: frontend/lib/prediction.ts
import { request } from "@/lib/api"

export interface PredictRequest {
  symptoms: string[]
}

export interface PredictItem {
  disease: string
  confidence: number
  probability: number
  similarity: number
  final_score: number
  matched_symptoms: string[]
  missing_symptoms: string[]
}

export interface PredictResponse {
  predictions: PredictItem[]
}

export async function predictSymptoms(payload: PredictRequest): Promise<PredictResponse> {
  return request<PredictResponse>("/predict", {
    method: "POST",
    body: JSON.stringify(payload),
  })
}
